test/unit/profiles/profile_context_test.rb

```diff
@@ -173,6 +174,15 @@ def load(call)
         get_checks.length.must_equal 1
         get_checks[0][1][0].resource_skipped.must_equal 'Skipped control due to only_if condition.'
       end
+
+      it 'doesnt extend into other control files' do
+        profile.load_control_file(if_false + control)
+        profile.load_control_file(control_2)
+        check_1 = get_checks
+        check_2 = Inspec::Rule.prepare_checks(profile.rules.values[1])
```

Comment:
Can we add an optional parameter to get_checks to have it pull values other than the first? That way we're not defining this logic in multiple places.

Reply:
I agree! Fixed :smile: